qiskit/primitives/base/base_estimator.py

```diff
@@ -125,6 +126,16 @@ def __init__(
         self._observables = []
         self._parameters = []
         super().__init__(options)
+        # Raise warning if subclass doesn't override `run` method in anticipation of
+        # it becoming an `abstractmethod`
+        if self.run.__qualname__ == BaseEstimator.run.__qualname__:
+            warnings.warn(
+                "The `BaseEstimator.run` method being callable is deprecated as of Qiskit 0.46"
+                " and will be converted to an abstractmethod in Qiskit 1.0. Subclasses should"
+                " implement a `run` with the same signature and its own validation directly.",
+                DeprecationWarning,
+                stacklevel=2,
```

Comment:
This isn't something that's user actionable. As an end user if you do:

```python
from qiskit_ibm_runtime.primitives import Estimator

Estimator()
```

you'll see the deprecation warning for this, but you don't control the implementation or how `qiskit-ibm-runtime` implemented their `Estimator` class.